src/config/external_libraries.rs

```diff
@@ -0,0 +1,115 @@
+use toml;
+
+use super::error::*;
+use nameutil::crate_name;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ExternalLibrary {
+    pub namespace: String,
+    pub crate_name: String,
+}
+
+pub fn read_external_libraries(toml: &toml::Value) -> Result<Vec<ExternalLibrary>> {
+    let mut external_libraries = match toml.lookup("options.external_libraries") {
+        Some(a) => {
+            try!(a.as_result_vec("options.external_libraries"))
+                .iter()
+                .filter_map(|v| v.as_str().map(String::from))
+                .map(|namespace| {
+                    ExternalLibrary {
+                        crate_name: crate_name(&namespace),
+                        namespace: namespace,
+                    }
+                })
+                .collect()
+        }
+        None => Vec::new(),
+    };
+    let custom_libs = toml.lookup("external_libraries").and_then(|v| v.as_table());
+    if let Some(custom_libs) = custom_libs {
+        for custom_lib in custom_libs {
+            let crate_name = custom_lib.0;
+            if let Some(namespace) = custom_lib.1.as_str() {
+                let lib = ExternalLibrary {
+                    namespace: namespace.to_owned(),
+                    crate_name: crate_name.clone(),
+                };
+                external_libraries.push(lib);
+            } else {
+                bail!("For external library \"{}\" namespace must be string", crate_name);
+            }
+
+        }
+    }
+
+    Ok(external_libraries)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use toml;
+
+    fn toml(input: &str) -> toml::Value {
+        let value = toml::from_str(&input);
+        assert!(value.is_ok());
+        value.unwrap()
+    }
+
+    #[test]
+    fn test_read_external_libraries() {
+        let toml = toml(
+            r#"
+[options]
+external_libraries = [
+   "GLib",
+   "Gdk",
+   "GdkPixbuf",
+]
+
+[external_libraries]
+coollib="CoolLib"
+other-lib="OtherLib"
```

Comment:
Are all possible crate names valid toml keys? Or are there crate names that wouldn't be allowed here?